fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,234 @@
+        log.entering(this.getClass().getName(), "init");
+
+        SSLContext ctx = null;
+
+        try {
+            this.channelName = channelName;
+            if (log.isLoggable(Level.FINER)) {
+                log.finer("ClusterId: " + clusterId);
+                log.finer("Channel name: " + channelName);
+                log.finer("ClientId: " + clientId);
+                log.finer("Servers: " + servers);
+            }
+
+            // Make sure that the properties file contains the expected properties.
+            if (clusterId == null || channelName == null || clientId == null || servers == null || servers.length() == 0) {
+                throw new IllegalStateException("Config property missing from the NATS connection properties.");
+            }
+
+            if (Boolean.parseBoolean(tlsProps.getProperty("useTLS"))) {
+                // Make sure that the tls properties are set.
+                if (tlsProps.getProperty("truststore") == null || tlsProps.getProperty("truststore-pw") == null || 
+                    tlsProps.getProperty("keystore") == null || tlsProps.getProperty("keystore-pw") == null) {
+                    throw new IllegalStateException("TLS config property missing from the NATS connection properties.");
+                }
+
+                ctx = SSLUtils.createSSLContext(tlsProps);
+            }
+
+            // Create the NATS client connection options
+            io.nats.client.Options.Builder builder = new io.nats.client.Options.Builder();
+            builder.maxReconnects(-1);
+            builder.connectionName(channelName);
+            builder.servers(servers.split(","));
+            if (ctx != null) {
+                builder.sslContext(ctx);
+            }
+            io.nats.client.Options natsOptions = builder.build();
+
+            // Create the NATS connection and the streaming connection
+            Connection nc = Nats.connect(natsOptions);
+            Options streamingOptions = new Options.Builder().natsConn(nc).build();
+            sc = NatsStreaming.connect(clusterId, clientId, streamingOptions);
+
+            // Create the publish callback
+            acb = new AckHandler() {
+                @Override
+                public void onAck(String nuid, Exception ex) {
+                    log.finer("Received ACK for guid: " + nuid);
+                    if (ex != null) {
+                        log.log(Level.SEVERE, "Error in server ack for guid " + nuid + ": " + ex.getMessage(), ex);
+                    }
+                }
+            };
+
+            // Register this NATS implementation as a "subscriber" with our Notification Service.
+            // This means that our "notify" method will be called when the server publishes an event.
+            service.subscribe(this);
+            log.info("Initialized NATS publisher for channel '" + channelName + "' using servers: " + servers + ".");
```

Comment:
```suggestion
            log.info("Initialized NATS publisher for channel '" + channelName + "' using servers: '" + servers + "'.");
```

you bracket the logged channelName, best to bracket both